datarobot_batch_scoring/batch_scoring.py

```diff
@@ -78,29 +78,61 @@ def slow_to_csv_chunk(data, header):
     return buf.getvalue()
 
 
-class FastReader(object):
-    """A reader that only reads the file in text mode but not parses it. """
+class Recoder:
```

Comment:
it re-encodes the python2 byte stream from whatever it was, to utf-8. The Python 2 csv module can only accept ascii or utf-8.